Include/Rover/ListTrial.hpp

```diff
@@ -68,17 +69,17 @@ namespace Rover {
   template<typename S>
   template<typename Begin, typename End>
   void ListTrial<S>::insert(Begin begin, End end) {
-    m_samples.insert(m_samples.end(), begin, end);
+    std::copy(begin, end, std::back_inserter(m_samples));
```

Comment:
Use ```move``` instead of ```copy```.

Reply:
But is it always correct? I may want to do something like this (and I do so in unit tests already actually):
```
auto t1 = vector<Sample<int, int>>();
t1.push_back(...);
auto t2 = ListTrial<Sample<int, int>>();
t2.insert(t1.begin(), t1.end());
// re-use t1
```
Furthermore, insertion by using TrialIterators of another trial may not work as TrialIterator is in fact a constant iterator.

Reply:
Good observation. Look into this issue, my understanding is that using ```move``` will only move objects when it's the "correct" thing to do, otherwise the move is simply a copy. For example, if you change the implementation to use ```move``` instead of ```copy``` then the following snippet of code will copy:

    std::vector<SomeSample> s;
    ...
    SomeTrial t = ...;
    t.insert(s.begin(), s.end()); // This will only copy, it won't move.

However, if the following snippet will move instead of copy.

    std::vector<SomeSample> s;
    ...
    SomeTrial t = ...;
    t.insert(std::make_move_iterator(s.begin()), std::make_move_iterator(s.end())); // This will move.

The issue is if you use ```std::copy``` in the implementation, then a copy is always made, even if we pass it "move" iterators.

But, please confirm my suspicion because there's a good chance I'm wrong.

Reply:
According to my tests and some stackoverflow guys, it is the opposite:
`std::move(s.begin(), s.end());` will always move as long as there is a move constructor for the underlying type;
`std::copy(s.begin(), s.end());` will take copies as expected;
`std::copy(std::make_move_iterator(s.begin()), std::make_move_iterator(s.end()));` will move as long as there is a move constructor for the underlying type (same as with `std::move`, but now I get a chance to choose since I can pass move iterators from outside).